src/Docnet.Core/DocLib.cs

```diff
@@ -123,14 +122,31 @@ public byte[] Split(byte[] bytes, int pageFromIndex, int pageToIndex)
         {
             Validator.CheckBytesNullOrZero(bytes, nameof(bytes));
 
-            Validator.CheckNotLessThanZero(pageFromIndex, nameof(pageFromIndex));
-            Validator.CheckNotLessThanZero(pageToIndex, nameof(pageToIndex));
-
-            Validator.CheckNotGreaterThan(pageFromIndex, pageToIndex, nameof(pageFromIndex), nameof(pageToIndex));
+            _pageRangeValidator.ValidatePageIndices(pageFromIndex, pageToIndex, nameof(pageFromIndex), nameof(pageToIndex));
 
             return _editor.Split(bytes, pageFromIndex, pageToIndex);
         }
 
+        /// <inheritdoc />
+        public byte[] Split(string filePath, string pageRange)
+        {
+            Validator.CheckFilePathNotNull(filePath, nameof(filePath));
+
+            _pageRangeValidator.ValidatePageNumbers(pageRange, nameof(pageRange));
+
+            return _editor.Split(filePath, pageRange);
+        }
+
+        /// <inheritdoc />
+        public byte[] Split(byte[] bytes, string pageRange)
+        {
+            Validator.CheckBytesNullOrZero(bytes, nameof(bytes));
+            _pageRangeValidator.ValidatePageNumbers(pageRange, nameof(pageRange));

```

Comment:
Just keep with static validators for now, maybe validation should have been a non static class but as it is it is good enough and it is better to be consistent. There is no reason to be afraid of static methods in circumstances like these. I would like to hear what you think and why you decided to do it this way.

Reply:
As mentioned above, usually I avoid static methods. It happened many times, when I required object's mock, but could not do that.

Reply:
Changed